### DISCUSSION TITLE
Give me access to the oggvorbis import window's audio visualizer

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#6385] [GUI]
### Describe the project you are working on

Rhythm game

### Describe the problem or limitation you are having in your project

I don't want to have to code it myself in GDScript

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It will give me access to the audio visualizer

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I will call a function to get the data.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

The import windows are core. I'm sure there's an audio visualizer in the code somewhere, I just can't find it.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

I feel this is better done with an add-on. Drawing a waveform doesn't require that much code, but you often want a lot of control over it (for artistic purposes). The editor's audio visualizer has a very specific goal – it's purely functional, with no artistic intent.

**Comment by ghost** [comment]
I see. I guess I will have to read the ogg docs myself then. 😭

**Reply by bossoverdose** [reply]
There are some tutorials on youtube with code source, maybe help you

[How To Make a BOTW Style Audio Visualizer in Godot in 11 Minute](https://www.youtube.com/watch?v=rNiWlOPxMhc)
[Godot Spectrum analyzer - Cool new audio feature in Godot 3.2](https://www.youtube.com/watch?v=066GA4dHqy4)
